main.go

```diff
@@ -111,8 +116,15 @@ func run(c *cli.Context) error {
 		go func() { startchan <- struct{}{} }()
 		file := c.String("file")
 		configchan := make(chan bool)
+		// XXX Unwatchable Puppet sources should get events from a timer,
+		// with the same interval as the Puppet agent
+		// (hint: puppet config print runinterval)
 		if !c.Bool("no-watch") {
-			configchan = ConfigWatch(file)
+			if c.IsSet("file") {
+				configchan = ConfigWatch(file)
+			} else if strings.HasSuffix(c.String("puppet"), ".pp") {
+				configchan = ConfigWatch(c.String("puppet"))
```

Comment:
Here are some other points that might convince you. Let me know what you think...

1) If we're doing the .pp file watching, then this is only going to work when we have local puppet files we're running. For the other two cases (using the puppet server, or inline code) this doesn't happen, so the behaviour is unique to one use case, and not consistent elsewhere.

2) Reminder, even though we'll generate a (possibly) new graph every 30 min, it will only be different if we've updated the code, or the facts. Once it's running, you still get the dynamic event based and parallel goodness.